Yoyo

yoyek opened the pull request:
Added nesting info on token page along with basic site styling.

yoyek committed:
Add moonbirds component with nest details on token page

yoyek committed:
Apply new styling for navbar and token page (using sass)

0xD8FF commented on components/dbs/MoonbirdCard.tsx:
```diff
@@ -0,0 +1,142 @@
+  const owner =
+    token?.kind === 'erc1155' && details?.market?.floorAsk?.maker
+      ? details?.market?.floorAsk?.maker
+      : token?.owner
```
this could probably be owner = token?.owner
for a moonbird token, token.kind will always be erc721.

if you need to check that the token is a moonbird token you could look at the contract address.

yoyek replied:
This is the code I copied from token/Owner.tsx from the original code, as I
incorporated owner inside MoonbirdCard.tsx.

The collection is checked on /pages/[contract]/[tokenId]/index.tsx, so our
component is shown only on that collection, otherwise the Owner.tsx is
shown (as it was originally)

So if i understand correctly, const owner = token.owner should be enough,
right?

0xD8FF commented on components/dbs/MoonbirdCard.tsx:
```diff
@@ -0,0 +1,142 @@
+  moonbird?: Object
+}
+
+const MoonbirdCard: FC<Props> = ({ token, moonbird }) => {
```
we should also include information about if the moonbird is banned on opensea. we would need the bannedonopensea prop for that.

yoyek replied:
Right.

0xD8FF commented on pages/[contract]/[tokenId]/index.tsx:
```diff
@@ -153,7 +155,11 @@ const Index: NextPage<Props> = ({ collectionId, tokenDetails }) => {
       <div className="col-span-full mb-4 space-y-4 px-2 md:col-span-4 md:col-start-5 lg:col-span-5 lg:col-start-7 lg:px-0 2xl:col-span-5 2xl:col-start-7 3xl:col-start-9 4xl:col-start-11">
-        <Owner details={token} bannedOnOpenSea={bannedOnOpenSea} />
+        {moonbird?.nesting ? (
+          <MoonbirdCard token={token.token} moonbird={moonbird} />
```
lets include the bannedonopensea information for moonbirds as well. might want to pass in the bannedonopensea prop for that.

you also might want to call the token prop for MoonbirdCard 'details' to keep it consistent with the Owner component.

yoyek replied:
About naming the props "details" instead of "token". I see one consistency in original code - when {token} is propagaded the property is "details", but when {token.token} is propagated its called "token".

Two examples:
```
<Owner details={token} bannedOnOpenSea={bannedOnOpenSea} />
<TokenAttributes token={token?.token} collection={collection} />
```

0xD8FF commented on pages/[contract]/[tokenId]/index.tsx:
```diff
@@ -229,7 +235,16 @@ export const getStaticProps: GetStaticProps<{
+  // Fetch moonbird details
+  let moonbird: Object = null
+  if (collectionId === '0x23581767a106ae21c074b2276d25e5c3e136a68b' && data?.tokens?.[0]?.token) {
+    const mbHref = 'https://birdwatching.moonbirds.xyz/moonbirds/' + data?.tokens?.[0]?.token?.tokenId
```
can you use the tokenId const that's set above on line 193? then you could do something like:
const mbHref = 'https://birdwatching.moonbirds.xyz/moonbirds/' + tokenId

0xD8FF commented on pages/[contract]/[tokenId]/index.tsx:
```diff
@@ -229,7 +235,16 @@ export const getStaticProps: GetStaticProps<{
+    const mbRes = await fetch(mbHref)
+    const mbData = await mbRes.json()
+    moonbird = mbData.moonbird
+  }
```
instead of using let could it be safer to do something like this:

```
const mbUrl = 'https://birdwatching.moonbirds.xyz/moonbirds/' + tokenId
const mbRes = await fetch(mbHref)
const moonbird: Object | undefined = collectionId === '0x23581767a106ae21c074b2276d25e5c3e136a68b' && await mbRes.json()
```

yoyek replied:
It would fire fetch() regardless if it's right collection, and as we try not
to hit API so often, I wrapped it all in "if" statement, that's why I'm using let on
moonbird var.

0xD8FF reviewed (commented):
didn't run this but it makes sense and looks good. let's include the banned on opensea info for the moonbirds page. i also added a few comments you might want to consider.

yoyek commented:
On Thu, Oct 20, 2022 at 2:00 PM uint160xffff ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> didn't run this but it makes sense and looks good. added a few comments
> you might want to consider.
> ------------------------------
>
> In components/dbs/MoonbirdCard.tsx
> <https://github.com/otint/dbs-market/pull/3#discussion_r1000488342>:
>
> > +  const owner =
> +    token?.kind === 'erc1155' && details?.market?.floorAsk?.maker
> +      ? details?.market?.floorAsk?.maker
> +      : token?.owner
>
> this could probably be owner = token?.owner
> for a moonbird token, token.kind will always be erc721.
>
> if you need to check that the token is a moonbird token you could look at
> the contract address.
>
This is a code I copied from token/Owner.tsx from the original code, as I
incorporated owner inside MoonbirdCard.tsx.
The collection is checked on /pages/[contract]/[tokenId]/index.tsx, so our
component is shown only on that collection, otherwise the Owner.tsx is
shown (as it was originally)

So if i understand correctly, const owner = token.owner should be enough,
right?


> ------------------------------
>
> In components/dbs/MoonbirdCard.tsx
> <https://github.com/otint/dbs-market/pull/3#discussion_r1000495237>:
>
> > @@ -0,0 +1,142 @@
> +import { FC } from 'react'
> +import EthAccount from 'components/EthAccount'
> +import Link from 'next/link'
> +import { TokenDetails } from 'types/reservoir'
> +import { formatDistanceToNow, formatDistance, parseISO, format } from 'date-fns'
> +
> +type Props = {
> +  token?: TokenDetails
> +  moonbird?: Object
> +}
> +
> +const MoonbirdCard: FC<Props> = ({ token, moonbird }) => {
>
> we should also include information about if the moonbird is banned on
> opensea. we would need the bannedonopensea prop for that.
>
------------------------------
>
> In pages/[contract]/[tokenId]/index.tsx
> <https://github.com/otint/dbs-market/pull/3#discussion_r1000496663>:
>
> > @@ -153,7 +155,11 @@ const Index: NextPage<Props> = ({ collectionId, tokenDetails }) => {
>          </div>
>        </div>
>        <div className="col-span-full mb-4 space-y-4 px-2 md:col-span-4 md:col-start-5 lg:col-span-5 lg:col-start-7 lg:px-0 2xl:col-span-5 2xl:col-start-7 3xl:col-start-9 4xl:col-start-11">
> -        <Owner details={token} bannedOnOpenSea={bannedOnOpenSea} />
> +        {moonbird?.nesting ? (
> +          <MoonbirdCard token={token.token} moonbird={moonbird} />
>
> lets include the bannedonopensea information for moonbirds as well. might
> want to pass in the bannedonopensea prop for that.
>
> you also might want to call the token prop for MoonbirdCard 'details' to
> keep it consistent with the Owner component.
>

Ok, need to look at it, as I saw that property in other components, did not
investigated further.


> ------------------------------
>
> In pages/[contract]/[tokenId]/index.tsx
> <https://github.com/otint/dbs-market/pull/3#discussion_r1000500870>:
>
> > @@ -229,7 +235,16 @@ export const getStaticProps: GetStaticProps<{
>      }
>    }
>
> +  // Fetch moonbird details
> +  let moonbird: Object = null
> +  if (collectionId === '0x23581767a106ae21c074b2276d25e5c3e136a68b' && data?.tokens?.[0]?.token) {
> +    const mbHref = 'https://birdwatching.moonbirds.xyz/moonbirds/' + data?.tokens?.[0]?.token?.tokenId
>
> can you use the tokenId const that's set above on line 193? then you could
> do something like:
> const mbHref = 'https://birdwatching.moonbirds.xyz/moonbirds/' + tokenId
>
Yup, right.


> ------------------------------
>
> In pages/[contract]/[tokenId]/index.tsx
> <https://github.com/otint/dbs-market/pull/3#discussion_r1000532219>:
>
> > +  let moonbird: Object = null
> +  if (collectionId === '0x23581767a106ae21c074b2276d25e5c3e136a68b' && data?.tokens?.[0]?.token) {
> +    const mbHref = 'https://birdwatching.moonbirds.xyz/moonbirds/' + data?.tokens?.[0]?.token?.tokenId
> +    const mbRes = await fetch(mbHref)
> +    const mbData = await mbRes.json()
> +    moonbird = mbData.moonbird
> +  }
>
> instead of using let could it be safer to do something like this:
>
> const mbUrl = 'https://birdwatching.moonbirds.xyz/moonbirds/' + tokenId
> const mbRes = await fetch(mbHref)
> const moonbird: Object | undefined = await mbRes.json()
>
> It would fire fetch regardless if it's right collection, and as we try no
to hit API so often I wrapped it all in if, that's why I'm using let on
moonbird.



>  —
> Reply to this email directly, view it on GitHub
> <https://github.com/otint/dbs-market/pull/3#pullrequestreview-1149043713>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABT6OQHTZPPNA5DHTT3RN3WEEX65ANCNFSM6AAAAAARJSUSYY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

yoyek committed:
Add alert of banned user on OpenSea to MoonbirdCard + minor fixes

yoyek committed:
Fix class attribute in MoonbirdCard

otint commented:
@yoyek getting a build error
```

$ next build
--
17:02:13.682 | info  - Loaded env from /vercel/path0/.env.production
17:02:13.938 | info  - Checking validity of types...
17:02:26.698 | Failed to compile.
17:02:26.699 |  
17:02:26.699 | ./components/dbs/MoonbirdCard.tsx:14:34
17:02:26.699 | Type error: Cannot find name 'details'.
17:02:26.699 |  
17:02:26.699 | 12 \| const MoonbirdCard: FC<Props> = ({ token, moonbird }) => {
17:02:26.700 | 13 \|   const owner =
17:02:26.700 | > 14 \|     token?.kind === 'erc1155' && details?.market?.floorAsk?.maker
17:02:26.700 | \|                                  ^
17:02:26.700 | 15 \|       ? details?.market?.floorAsk?.maker
17:02:26.700 | 16 \|       : token?.owner
17:02:26.700 | 17 \|
17:02:26.735 | error Command failed with exit code 1.
17:02:26.736 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
17:02:26.760 | Error: Command "yarn run build" exited with 1

```

yoyek commented:
Check the newest commits, specially [this one](https://github.com/otint/dbs-market/pull/3/commits/92949e734d10fffae13d4045565a11d79e022224)
This is no longer in codebase:
`token?.kind === 'erc1155' && details?.market?.floorAsk?.maker`

yoyek committed:
Fix typing errors on build

otint commented:
@yoyek still getting a build error, see here:

<img width="946" alt="image" src="https://user-images.githubusercontent.com/258621/197652371-f9a953cd-635d-47cd-9bd1-73cb333ec0bb.png">

yoyek commented:
I don't get it.
It's using the old code to build.

This is no longer in code base for like 5 days now.

![image](https://user-images.githubusercontent.com/212794/197718247-2b9b0032-7873-4e37-84be-45ca5eb35566.png)

[See this commit](https://github.com/otint/dbs-market/pull/3/commits/92949e734d10fffae13d4045565a11d79e022224)

yoyek committed:
Merge branch 'main' of https://github.com/otint/dbs-market into yoyo

# Conflicts:
#	package.json
#	pages/[contract]/[tokenId]/index.tsx
